src/base.rs

```diff
@@ -30,72 +33,121 @@ impl GeneralTaxonomy {
     /// calls this), but it's possible that a user might want to save
     /// memory/start-up time so it's theoretically possible to use this
     /// struct without running this.
-    fn index(mut self, tax_id: bool, children: bool) -> Self {
-        if tax_id && self.tax_id_lookup == None {
-            let mut tax_id_lookup = HashMap::with_capacity(self.tax_ids.len());
-            for (ix, tax_id) in self.tax_ids.iter().enumerate() {
-                tax_id_lookup.insert(tax_id.clone(), ix);
-            }
-
-            self.tax_id_lookup = Some(tax_id_lookup);
+    fn index(&mut self) {
+        let mut tax_id_lookup = HashMap::with_capacity(self.tax_ids.len());
+        for (ix, tax_id) in self.tax_ids.iter().enumerate() {
+            tax_id_lookup.insert(tax_id.clone(), ix);
```

Comment:
NCBI's is ~2 million nodes so it'd be nice to remove the cloning, but I couldn't figure out a good way to make the `HashMap<&str, IntTaxId>` without getting yelled at about the lifetimes of the keys (since they're self-referential inside the struct).

Reply:
Good point. I could have a look later but I would need to read the whole crate really so unlikely to happen for this review :p

Reply:
I really like this idea, but I don't think it'll change the external API so I made an issue for it at #19